src/repo/secret/infisical.go

```diff
@@ -0,0 +1,38 @@
+package secret
+
+import (
+	"sync"
+
+	infisical "github.com/infisical/go-sdk"
+)
+
+type Secret struct {
+	client infisical.InfisicalClientInterface
+	mu     sync.Mutex
+}
+
+func NewSecret(secret *Secret) (*Secret, error) {
+	secret.mu.Lock()
+	client := infisical.NewInfisicalClient(infisical.Config{
+		SiteUrl: "https://app.infisical.com",
+	})
+	// use INFISICAL_UNIVERSAL_AUTH_CLIENT_ID, INFISICAL_UNIVERSAL_AUTH_CLIENT_SECRET env
```

Comment:
this values INFISICAL_UNIVERSAL_AUTH_CLIENT_ID and INFISICAL_UNIVERSAL_AUTH_CLIENT_SECRET should come as params to NewSecret func